app/src/main/cpp/SJCL.cpp

```diff
@@ -176,7 +273,96 @@ char* SJCL::decrypt(string sjcl_json, string key) {
     free(cyphertext);
     free(data);
     free(derived_key);
-//    free(food);
+
+    return ret;
+}
+
+char* SJCL::encrypt(char* plaintext, const string& key) {
+    int plaintext_len = strlen(plaintext);
+    int iv_len = 13;    // use 13 because 15-lol (with initial lol=2) is hardcoded in the decryptccm implementation
+
+    // strange iv_len calculation due to the decryptccm implementation
+    if (plaintext_len >= 1<<16) iv_len--;
+    if (plaintext_len >= 1<<24) iv_len--;
+
+    int salt_len = 12;  //can I use a random number here?
```

Comment:
short answer: yes

long answer:
it should have a minimum length and a variable max, also some ceiling to not go too crazy, if you wish to make it random in length

Reply:
I would leave it that way for now. Added it to my todo.